protocol/p2p/src/core/peer.rs

```diff
@@ -13,14 +13,16 @@ pub struct PeerProperties {
     pub time_offset: i64,
 }
 
-#[derive(Debug)]
+#[derive(Debug, Eq, PartialEq, Hash)]
```

Comment:
I don't think this should be the default behaviour for checking equality, but maybe `filtered_eviction_set` should use `PeerId` instead of `Peer`? And then we can remove the `Eq, PartialEq, Hash` from here